### PR TITLE
Fix (llm): small fixes to LLM

### PR DESCRIPTION
**Pull request by Giuseppe5**
- Improve HQO implementation
- Support for MSE with groupwise quantization
- Add possibility to specify groupdim for groupwise weight quantization

### CODE REVIEW

File: src/brevitas/core/function_wrapper/shape.py [hunk 1]
```diff
@@ -165,6 +165,11 @@ def __init__(self, expanded_groupwise_shape, group_size, group_dim) -> None:
 
     @brevitas.jit.script_method
     def forward(self, x: torch.Tensor):
+        # This one is a bit tricky but we could end up here:
+        # - If we quantize the zero point, which will already have expanded shape matching the scale (although no padding, but we don't need the padding)
+        # - Groupwise HQO quantization, where weight will already have been padded and expanded
+        if len(x.shape) == len(self.expanded_groupwise_shape):
+            return x
```
Comment (nickfraser) [inline src/brevitas/core/function_wrapper/shape.py:172]: Weird stuff / comments like this make me wonder if we need to re-think our implementation.

**Review by nickfraser — APPROVED** [review]
LGTM!

File: src/brevitas/core/function_wrapper/shape.py (hunk 1 above)
Comment (nickfraser) [reply src/brevitas/core/function_wrapper/shape.py:172]: (but let's not block this release)

Comment (Giuseppe5) [reply src/brevitas/core/function_wrapper/shape.py:172]: Agreed

**Pull request merged**